### ISSUE TITLE
Native git integration

### ISSUE THREAD

**Issue by prmichaelsen** [a85/POSTMan-Chrome-Extension#3563]
## Feature Request
Our team uses Postman for quick sanity testing of our APIs and for populating data into various environments to allow us to test certain UI workflows. While we appreciate Postman’s cloud services for managing collections, we would much rather use tools like git to keep our tests and API calls with our central repo. Currently there is good support for CLI testing of collections with tools like newman. This works very well with git. However, importing and exporting our collections to and from Postman GUI every time we make test changes or checkout new changes in git is unwieldly and cumbersome. As Postman tests are text files, it would be very helpful if Postman could automatically read and write collections directly from a source directory without the need to import or export. Many of the other tools in our toolchain integrate seamlessly with git and it would be fantastic if Postman did as well.

**Comment by mccannt** [comment]
+1 
This would be huge for my company as well. Currently we have the import/export from our internal CVS each time we make changes.

**Comment by tbuttier** [comment]
I totally agree. It would be great to have filesystem integration so we could manage the changes with GIT directly.
Moreover, it would be appreciated if no changes were produced just for saving a file.

**Comment by surfsun** [comment]
+1

**Comment by maskaravivek** [comment]
+1 This would be useful for us as well :)

**Comment by ishepherd** [comment]
My #3910 got closed, but I don't think you should bother with `Native git integration`. Instead, refactor Postman to turn its native storage format into text files. Just like all other dev tools.

If you can pull this off, yes `git` will integrate easily, and so will `grep` and `jq` and the entire universe of file based tools. We will be able to write generators for Postman test collections and then tweak the result in the Postman UI. All kinds of things are possible :)

**Comment by Jillxm** [comment]
+1 Very important to us.  Assuming this feature would include Global variables it would allow source control/integrations for functions which need to be stored as variables.

**Comment by ishepherd** [comment]
Just make the native storage into text files.

We have perfectly good git clients, and so many other tools besides, which are all currently locked out of Postman, due to your legacy practice of keeping the collections in a private database.

**Comment by ishepherd** [comment]
While I'm on my soapbox: "text files", plural. The current export generates an entire collection as a single file, this makes large collections extremely difficult for git merging and GitHub pull requests.

**Comment by rohit-gohri** [comment]
This would really improve Postman as an API testing tool. The current workflow of exporting collections before committing and deleting & importing collections after changing branches is very cumbersome. Anyone have any better workflow suggestions?

Related old issues (linking for others who might stumble from Google): 
- https://github.com/postmanlabs/postman-app-support/issues/3475
- https://github.com/postmanlabs/postman-app-support/issues/730

**Comment by karlbrown-va** [comment]
Adding my vote for this. Seems sensible and quite useful to have a 'sync' function that would sync a given collection with its location on disk.

**Comment by anna-seredinskaya** [comment]
This is the number one feature we desperately need and are waiting for.

**Comment by vdespa** [comment]
It seems that Postman is taking the approach of integrating Git over the cloud services they offer and not directly from the app.

This involves practically reimplementing version control. This means that only a limited of Git providers will be included, as each needs to be supported individually. We are talking were mostly Github, GitLab and maybe just maybe Bitbucket). So I don't think it will be possible to use your own Git server for syncing Postman collections. 

I would have hoped Postman turns more into an IDE like IntelliJ that can work directly with local tools (like Git).

Currently, it looks like the dream of having a native Git integration is dead.

**Comment by Jillxm** [comment]
Valentin,

My postman collections only (not variables)  are currently If currently integrated with our private github server using access tokens to.

Does this email mean that the integrations using access tokens will no longer be supported?

Thanks,
Jill

**Comment by matt-ball** [comment]
Whilst not native, I would recommend looking at [Postman CLI](https://github.com/matt-ball/postman-cli) if you'd like to facilitate local development of requests, responses and scripts for Postman.

**Comment by vdespa** [comment]
@Jillxm - Your current GitLab setup will work just fine. This issue is about a native Git integration which is not the current approach.

**Comment by Rafnafnaf** [comment]
+1 This is so much of a problem for my team that we are considering changing API testing tool for another one which will work with git more easily.

**Comment by prmichaelsen** [comment]
Considering this is in the top 50 highest voted issues perhaps it warrants
a response from the team

**Comment by yosadchyi** [comment]
+1, it seems to be blocker for using postman as main API testing tool

**Comment by AjaySewradj** [comment]
How do you vote this ticket up?

We also desperately need file-system integration as described in all the posts above.
Any updates on this ticket?

**Comment by prmichaelsen** [comment]
> How do you vote this ticket up?
> 
> We also desperately need file-system integration as described in all the posts above.
> Any updates on this ticket?

It’s been a top issue for years and no official response. It would seem this is not a priority for postman. I would consider a product that’s open source as an alternative.

**Comment by tyomo4ka** [comment]
There is 1.8k issues in this repo.
I guess someone from the team needs to do a bit of grooming :)

I will just make a desperate attempt to bump up this one again.

**Comment by arlemi** [comment]
Hey everyone! We've added native Git support for Bitbucket and GitHub in Postman v9. Please[ check this blog post](https://blog.postman.com/announcing-new-postman-api-platform/) for more info, and update your app to try it out!

**Comment by prmichaelsen** [comment]
holy shit 

![it's over. it's done.](https://i.imgur.com/oXFtpr2.jpeg)

**Comment by AjaySewradj** [comment]
> Hey everyone! We've added native Git support for Bitbucket and GitHub in Postman v9. Please[ check this blog post](https://blog.postman.com/announcing-new-postman-api-platform/) for more info, and update your app to try it out!

I'm only using 'Collections' in Postman. It doesn't seem that git support is enabled for the Collections. So I'm still stuck exporting the Collections to the filesystem on a regular basis.

So the original issue has not been fixed

**Comment by prmichaelsen** [comment]
just discovered https://hoppscotch.io/

it's an open source alternative to postman. get rekt postman.

**Comment by prmichaelsen** [comment]
@AjaySewradj you may be interested in checking out this open source alternative chartered in aug '19: https://hoppscotch.io/

"too little, too late" as far as the implementation of this feature goes. i'm always excited to bail on greedy closed source projects ^-^

**Comment by shashankawasthi88** [comment]
@prmichaelsen @AjaySewradj @tyomo4ka and others, I am reopening this ticket. We have started working on integration with local file systems. We want to understand your local development workflows. Feel free to schedule a call with us using [this](https://calendly.com/shashank-awasthi/local-git-feedback) link or feel free to drop me a mail at shashank.awasthi@postman.com describing your workflows.

**Comment by prmichaelsen** [comment]
@shashankawasthi88 here's an example of what I use today and how it works, which is guiding my workflow today: https://blog.bitsrc.io/vs-codes-rest-client-plugin-is-all-you-need-to-make-api-calls-e9e95fcfd85a

**Comment by shashankawasthi88** [comment]
The postman v10 now allows you to perform all the git workflows from Postman directly. Follow the [blogpost](https://blog.postman.com/announcing-postmans-redesigned-api-builder-with-native-git-support/) to know more.

**Comment by a85** [comment]
We have been working on a much better version of this request to satisfy the original intent and make Postman git-compatible. This will eventually supersede all existing Git integrations on the product. Postman workspaces that allow for pushing/pulling from the filesystem while also bringing in git-branch awareness that is useful for CI and pull request review workflows.

You can see it in action here: 

https://github.com/user-attachments/assets/7d29c5ff-ff2e-4c0f-a711-9499650cee92

Our engineers are using this internally and see huge gain in productivity in their workflow with the ability to tie collections, spec, and environment file changes tied to code commits.

**Comment by vdespa** [comment]
Congrats @a85 and team - this is fantastic news 🎉🥳

**Comment by a85** [comment]
Hello all, The integration I shared in my earlier video is rolling out to users in beta. We hope to GA this soon. Your feedback will be helpful here.

**Comment by shashankawasthi88** [comment]
Hi all,

As of **[Postman v12](https://www.postman.com/downloads/)**, we now support Native Git Integration.
You can connect a workspace directly to a local Git repository and work against your filesystem. Collections and related artifacts are stored as files, enabling standard Git workflows (branching, diffs, PRs, CI/CD) without manual export/import.
Docs: https://learning.postman.com/docs/agent-mode/native-git
Announcement: https://blog.postman.com/new-postman-is-here/
Feel free to try it out and share feedback.

https://github.com/user-attachments/assets/8fbc0933-1095-4bbc-98c8-f1d09cc84189
